charm/src/actions.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""HTTPRequest LEGO provider charm actions."""
+
+# pylint: disable=protected-access
+
+import logging
+import secrets
+
+import ops
+import xiilib.django
+
+logger = logging.getLogger(__name__)
+
+
+class NotReadyError(Exception):
+    """Exception thrown when needed resources are not ready."""
+
+
+class Observer(ops.Object):
+    """Charm actions observer."""
+
+    def __init__(self, charm: xiilib.django.Charm):
+        """Initialize the observer and register actions handlers.
+
+        Args:
+            charm: The parent charm to attach the observer to.
+        """
+        super().__init__(charm, "actions-observer")
+        self.charm = charm
+
+        charm.framework.observe(charm.on.create_user_action, self._create_or_update_user)
+        charm.framework.observe(charm.on.allow_domains_action, self._allow_domains)
+        charm.framework.observe(charm.on.revoke_domains_action, self._revoke_domains)
+        charm.framework.observe(charm.on.list_domains_action, self._list_domains)
```

Comment:
Is the function naming convention correct here? Should the functions be called `self._on_create_or_update_user`?